docs/resources/deployment.md

```diff
@@ -0,0 +1,854 @@
+---
+page_title: "Elastic Cloud: ec_deployment Resource"
+description: |-
+  Provides an Elastic Cloud deployment resource, which allows deployments to be created, updated, and deleted.
+---
+
+# Resource: ec_deployment
+
+Provides an Elastic Cloud deployment resource, which allows deployments to be created, updated, and deleted.
+
+~> **Note on Elastic Stack versions** Using a version prior to `6.6.0` is not supported.
+
```

Comment:
The note on traffic filters from the deleted `ec_deployment.md` is omitted now. Does it make sense to include it until the relevant PR is merged?